### PR TITLE
MAINT: Use explicitly public APIs

### PR DESCRIPTION
**Pull request by larsoner**
In https://github.com/mne-tools/mne-python/pull/11903 we're trying to make some mne/io stuff private. This changes mne-nirs to use the explicitly public API

Also includes a number of fixes for latest NumPy/SciPy/Pandas

### CODE REVIEW

File: examples/general/plot_10_hrf_simulation.py [hunk 1]
```diff
@@ -98,20 +98,21 @@ def print_results(glm_est, truth):
...
-plt.plot(raw.times, glm_est.theta()[0][0] * design_matrix["A"].values * 1e6)
-plt.xlabel("Time (s)")
-plt.ylabel("Haemoglobin (uM)")
-plt.legend(["Clean Data", "Noisy Data", "GLM Estimate"])
+fig, ax = plt.subplots(constrained_layout=True)
+ax.plot(raw.times, raw_noise_free.get_data().T * 1e6)
+ax.plot(raw.times, raw.get_data().T * 1e6, alpha=0.3)
+ax.plot(raw.times, glm_est.theta()[0][0] * design_matrix["A"].values * 1e6)
+ax.set_xlabel("Time (s)")
+ax.set_ylabel("Haemoglobin (uM)")
```
Comment (drammock) [inline examples/general/plot_10_hrf_simulation.py:114]:
```suggestion
ax.set(xlabel="Time (s)", ylabel="Haemoglobin (μM)")
```

File: examples/general/plot_10_hrf_simulation.py [hunk 2]
```diff
@@ -138,12 +139,13 @@ def print_results(glm_est, truth):
...
```
Comment (drammock) [inline examples/general/plot_10_hrf_simulation.py:147]:
```suggestion
ax.set(xlabel="Time (s)", ylabel="Haemoglobin (μM)")
```

File: examples/general/plot_10_hrf_simulation.py [hunk 3]
```diff
@@ -171,11 +173,12 @@ def print_results(glm_est, truth):
...
-plt.plot(raw.times, raw.get_data().T * 1e6, alpha=0.3)
-plt.plot(raw.times, glm_est.theta()[0][0] * design_matrix["A"].values * 1e6)
-plt.xlabel("Time (s)")
-plt.ylabel("Haemoglobin (uM)")
-plt.legend(["Noisy Data", "GLM Estimate"])
+fig, ax = plt.subplots(constrained_layout=True)
+ax.plot(raw.times, raw.get_data().T * 1e6, alpha=0.3)
+ax.plot(raw.times, glm_est.theta()[0][0] * design_matrix["A"].values * 1e6)
+ax.set_xlabel("Time (s)")
+ax.set_ylabel("Haemoglobin (uM)")
```
Comment (drammock) [inline examples/general/plot_10_hrf_simulation.py:180]:
```suggestion
ax.set(xlabel="Time (s)", ylabel="Haemoglobin (μM)")
```

File: examples/general/plot_11_hrf_measured.py [hunk 4]
```diff
@@ -227,13 +227,14 @@
...
-plt.xlim(180, 300)
-plt.legend(["Stimulus", "Expected Response"])
-plt.xlabel("Time (s)")
-plt.ylabel("Amplitude")
+ax.plot(raw_intensity.times, s[:, 1])
+ax.plot(design_matrix['Tapping_Left'])
+ax.set_xlim(180, 300)
+ax.legend(["Stimulus", "Expected Response"])
+ax.set_xlabel("Time (s)")
+ax.set_ylabel("Amplitude")
```
Comment (drammock) [inline examples/general/plot_11_hrf_measured.py:237]:
```suggestion
ax.legend(["Stimulus", "Expected Response"])
ax.set(xlim=(180, 300), xlabel("Time (s)", ylabel="Amplitude")
```

**Review by drammock — COMMENTED** [review]
nitpicks (I like examples to show what we consider best practices)

File: examples/general/plot_11_hrf_measured.py [hunk 5]
```diff
@@ -227,13 +227,12 @@
...
-plt.plot(raw_intensity.times, s[:, 1])
-plt.plot(design_matrix['Tapping_Left'])
-plt.xlim(180, 300)
-plt.legend(["Stimulus", "Expected Response"])
-plt.xlabel("Time (s)")
-plt.ylabel("Amplitude")
+ax.plot(raw_intensity.times, s[:, 1])
+ax.plot(design_matrix['Tapping_Left'])
+ax.legend(["Stimulus", "Expected Response"])
+ax.set(xlim=(180, 300), xlabel("Time (s)", ylabel="Amplitude")
```
[the remark this answers is not in the document]
Comment (larsoner) [reply examples/general/plot_11_hrf_measured.py:235]:
```suggestion
ax.set(xlim=(180, 300), xlabel="Time (s)", ylabel="Amplitude")
```

**Pull request merged**